src/backend/catalog/heap.c

```diff
@@ -2995,6 +2996,9 @@ cookDefault(ParseState *pstate,
 							format_type_be(atttypid),
 							format_type_be(type_id)),
 					 errhint("You will need to rewrite or cast the expression.")));
+
+		if (handle_type_and_collation_hook)
```

Comment:
Would you mind explain the user case why we need to add this line and follow up with test cases ? It can help us to understand the priority and necessary to merge this change into main branch.

Reply:
I explain the reason why we need this fix in issue https://github.com/babelfish-for-postgresql/babelfish_extensions/issues/4069
I also would like add test scenarios but the JDBC tests are in babelfish repository. How can I add my test cases here?

Reply:
Here I created the failing test case in babelfish_repository
https://github.com/babelfish-for-postgresql/babelfish_extensions/pull/4123

Reply:
İt is already renamed and the tests are passing. https://github.com/babelfish-for-postgresql/babelfish_extensions/pull/4130